usersexercise/src/main/java/com/Services/Implementations/UserServiceImplementation.java

```diff
@@ -0,0 +1,120 @@
+package com.Services.Implementations;
+
+import com.ExceptionHandler.ApiException;
+import com.ExceptionHandler.ApiRequestException;
+import com.Model.Response.UserResponse;
+import com.Services.UserService;
+import com.dao.UserRepository;
+import com.dto.UserDto;
+import org.springframework.beans.BeanUtils;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+import com.Model.Users;
+import org.springframework.data.domain.Pageable;
+import java.util.List;
+import java.util.*;
+
+@Service
+public class UserServiceImplementation implements UserService {
+
+    private final UserRepository userrepository;
```

Comment:
stay consistent with your naming conventions. In your models your fields are all snake_case. Here it's all lowercase